rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -142,6 +142,17 @@ Player::Player(
     auto metadata = reader_->get_metadata();
     starting_time_ = std::chrono::duration_cast<std::chrono::nanoseconds>(
       metadata.starting_time.time_since_epoch()).count();
+    // If a non-default (positive) starting time offset is provided in PlayOptions,
+    // then add the offset to the starting time obtained from reader metadata
+    if (play_options_.start_offset < INT64_C(0)) {
```

Comment:
```suggestion
    if (play_options_.start_offset < 0) {
```

Reply:
Addressed in https://github.com/ros2/rosbag2/pull/931/commits/07543e0c770db06fbada46268105fd4eea1a9d3a.